apps/docs-snippets/src/guide/introduction/getting-started.test.ts

```diff
@@ -11,9 +11,10 @@ import {
  * @group browser
  */
 describe('Getting started', () => {
-  beforeAll(() => {
+  beforeAll(async () => {
     // Avoids using the actual network.
-    vi.spyOn(Provider, 'create').mockImplementationOnce(() => Provider.create(FUEL_NETWORK_URL));
+    const mockProvider = await Provider.create(FUEL_NETWORK_URL);
+    vi.spyOn(Provider, 'create').mockResolvedValue(mockProvider);
```

Comment:
Why did you change this?

Shouldn't the `mockResolvedValue()` method return a promise, as the mocked method `create` does?

Reply:
The rationale is that mocks should return the same value as the methods they're mocking.

I suppose the original `Provide.create` returns an _unresolved_ promise, and so should the mock. 🤔 


```ts
const provider = Provider.create(FUEL_NETWORK_URL); // <— no await here
vi.spyOn(Provider, 'create').mockResolvedValue(provider);

// or simply
vi.spyOn(Provider, 'create').mockResolvedValue(Provider.create(FUEL_NETWORK_URL));
```